bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -576,16 +579,36 @@ public String stringValue() {
             case TypeTags.STRING_TAG:
             case TypeTags.CHAR_STRING_TAG:
                 for (int i = 0; i < size; i++) {
-                    sj.add(bStringValues[i].getValue());
+                    sj.add(((BValue) (bStringValues[i])).informalStringValue(parent));
                 }
                 break;
             default:
                 for (int i = 0; i < size; i++) {
-                    sj.add(StringUtils.getStringValue(refValues[i]));
+                    if (refValues[i] == null) {
+                        sj.add("null");
+                    } else {
+                        BType type = TypeChecker.getType(refValues[i]);
+                        switch (type.getTag()) {
+                            case TypeTags.STRING_TAG:
```

Comment:
is it possible to move XML toString part to the SrtingUtil as well? this is same as in mapvalueimpl and detail in errorvalue

Reply:
I don't think, it is possible since toString value of a string and string in a array is different.